### PR TITLE
Add org.kde.gwenview (Gwenview)

### PR DESCRIPTION
**Pull request by Siosm**

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.gwenview

**Comment by Siosm** [comment]
Mention @tsdgeos as he currently is a maintainer of several KDE related packages on Flathub.

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

**Comment by tsdgeos** [comment]
Does File->open work for you? I get nothing

**Comment by Siosm** [comment]
Indeed, looks like there is a bug here.

**Comment by tsdgeos** [comment]
I don't know if you have devel skills or not, is this something you can debug or should we open a bug in bugs.kde.org about it? or maybe try to pull in @grulja ?

**Comment by grulja** [comment]
This is same problem as was recently in OBS (https://github.com/obsproject/rfcs/pull/21#issuecomment-619173461).

Basically the OpenFile() call is being rejected because of an empty filter name. You can see it here:
```
              struct {
                  string "surový obrázek Panasonic"
                  array [
                     struct {
                        uint32 1
                        string "image/x-panasonic-rw2"
                     }
                  ]
               }
               struct {
                  string ""
                  array [
                     struct {
                        uint32 1
                        string "image/x-samsung-srw"
                     }
                  ]
               }

```
I see two filter names missing, one is for "**image/x-nikon-nrw**" and the other **"image/x-samsung-srw**".

**Comment by Siosm** [comment]
I will make a patch.

**Comment by Erick555** [comment]
@Siosm did you had a chance to look at this? Recently there was [upstream commit](https://invent.kde.org/graphics/gwenview/-/commit/16f85088fcd61886aba9fef5417df6ce9ad89213) related to mimetypes. I don't know if this is enough to fix flatpak but at least it shows what need to be changed.

It would be nice to have gwenview flatpak available 😄 .

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

**Comment by Siosm** [comment]
Thanks @Erick555! Updated with the patch you mentioned and looks like this is working now!

**Comment by Siosm** [comment]
Failed on aarch64 build :/

**Comment by Siosm** [comment]
https://github.com/Exiv2/exiv2/issues/1352

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

File: org.kde.gwenview.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+    "id": "org.kde.gwenview",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
+    "sdk": "org.kde.Sdk",
+    "command": "gwenview",
+    "rename-icon": "gwenview",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
```
Comment (Erick555) [inline org.kde.gwenview.json:10]: This should be replaced with `fallback-x11`

**Comment by Siosm** [comment]
bot, build org.kde.gwenview

File: org.kde.gwenview.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    "command": "gwenview",
+    "rename-icon": "gwenview",
+    "finish-args": [
+        "--device=dri",
+        "--filesystem=host",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland"
+    ],
+    "cleanup": [
```
Comment (Erick555) [inline org.kde.gwenview.json:15]: Please add `/lib/cmake`.

Comment (Siosm) [reply org.kde.gwenview.json:15]: Thanks, fixed!

**Comment by tsdgeos** [comment]
-- The following OPTIONAL packages have not been found:
 * CFitsio, FITS format support, <https://heasarc.gsfc.nasa.gov/fitsio/fitsio.html>
 * KF5Kipi, Provides various image manipulation and export features, <https://phabricator.kde.org/source/kipi-plugins/>
 * KF5Baloo (required version >= 5.1.90), Desktop-wide semantic information support, <https://commits.kde.org/baloo>
 * KF5KDcraw, C++ interface around LibRaw library used to decode RAW picture files, <https://phabricator.kde.org/source/libkdcraw/>

Do you want to try to get some of those deps in? Or let's roll with this first and try to improve later?

**Comment by Siosm** [comment]
I would say let's start with this version and keep track of the missing pieces in an issue that I will work on as soon as possible.

**Comment by Siosm** [comment]
Thanks @barthalion!

While I got you here, could you do my account switch? (see https://invent.kde.org/teams/flathub/issues/-/issues/8)

Thanks in advance!

**Comment by barthalion** [comment]
Done.

**Comment by Siosm** [comment]
Looks good. You can now remove my old account (Siosm). Thanks!

**Comment by Siosm** [comment]
I also need access to https://github.com/flathub/org.freeorion.FreeOrion for travier

**Comment by Siosm** [comment]
Thanks!

**Pull request closed without merging**